lib/src/main/java/edu/stanford/futuredata/macrobase/analysis/classify/PredicateCubeClassifier.java

```diff
@@ -0,0 +1,160 @@
+package edu.stanford.futuredata.macrobase.analysis.classify;
+
+import edu.stanford.futuredata.macrobase.datamodel.DataFrame;
+import edu.stanford.futuredata.macrobase.util.MacrobaseException;
+
+import java.util.function.DoublePredicate;
```

Comment:
confused, why does this file show up twice?

Reply:
This code duplication is a ugly hack that makes the predicate classifier work with cube pipelines (since cube classifiers and classifiers are separate). An alternative proposal (thanks @kexinrong) would be to create a classifier class, and have all other classifiers (cube and not) just extend this. The cube classifiers would all need to add their own count columns. This might be ok since we only check for a certain number of classifiers in the cube/batch pipelines. We should discuss which works best once you're back!